src/core/filters/filterregistry.js

```diff
@@ -73,19 +72,14 @@ export default class FilterRegistry {
   }
 
   /**
-   * Gets the facet filter string to send in a search query.
-   * @returns {string}
+   * Gets the facet filters as an array of SimpleFilters to send to the answers-core.
+   *
+   * @returns {SimpleFilter[]}
    */
   getFacetFilterPayload () {
-    return JSON.stringify(this._getFacetFilterPayload());
-  }
-
-  _getFacetFilterPayload () {
-    const getFilters = fn => fn.getChildren().length
-      ? fn.getChildren().flatMap(getFilters)
-      : fn.getFilter();
-    const filters = this.getFacetFilterNodes().flatMap(getFilters);
-    return Facet.fromFilters(this.availableFieldIds, ...filters);
+    return this.getFacetFilterNodes()
+      .flatMap(filterNode => filterNode.children || filterNode)
```

Comment:
Can we use the `getChildren()` getter here? If possible, we should always use the getter instead of the property directly. That's safer.

Reply:
I think I remember trying that and I got different results, but totally agree, going to test again

Reply:
Ah, it was a combination of the "refreshing the page doesn't actually apply facets to the search" and that `getChildren()` is not falsy when there are no children (empty array). Resolved and updating to use `getChildren()` and `getFilters()`!